pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -93,7 +93,23 @@ func (r *Reconciler) gatherSelectedResource(placement string, selectors []fleetv
 		if err != nil {
 			return nil, err
 		}
-		resources = append(resources, objs...)
+		for _, obj := range objs {
+			uObj := obj.(*unstructured.Unstructured)
+			ri := fleetv1beta1.ResourceIdentifier{
+				Group:     obj.GetObjectKind().GroupVersionKind().Group,
+				Version:   obj.GetObjectKind().GroupVersionKind().Version,
+				Kind:      obj.GetObjectKind().GroupVersionKind().Kind,
+				Name:      uObj.GetName(),
+				Namespace: uObj.GetNamespace(),
+			}
+			if _, exist := resourceMap[ri]; exist {
```

Comment:
Just a nit: I know we might not have to deal with this very often but a case could occur that the same resource was being selected twice using different versions , and it would bypass the check.

Reply:
i.e., both versions are served on the hub cluster, and since only one version can be set as the storage version, the two refs are pointing to the same object

Reply:
I am not sure we can block two objects with different versions, if k8s doesn't block this.